snippets/deepFreeze.md

```diff
@@ -0,0 +1,38 @@
+### deepFreeze
+
+Deep freezes an object.
+
+Calls `Object.freeze(obj)` recursively on all unfrozen properties of obj that are `typeof` function or object.
+
+```js
+const deepFreeze = obj => {
+  Object.freeze(obj);
+
+  Object.getOwnPropertyNames(obj).forEach(function(prop) {
+    if (
+      obj.hasOwnProperty(prop) &&
+      obj[prop] !== null &&
+      (typeof obj[prop] === 'object' || typeof obj[prop] === 'function') &&
+      !Object.isFrozen(obj[prop])
+    ) {
+      deepFreeze(obj[prop]);
+    }
+  });
+
+  return obj;
```

Comment:
I think that most of this could be converted to a ternary operator, but readility might be terrible.

Reply:
> but readility might be terrible.

It would. This isn't very long either.

Reply:
If we go with `instanceof` as suggested maybe it can work with a ternary. I'll check and report back with reults.